bson/primitive/primitive.go

```diff
@@ -40,11 +40,32 @@ type Undefined struct{}
 // DateTime represents the BSON datetime value.
 type DateTime int64
 
+var _ json.Marshaler = DateTime(0)
+var _ json.Unmarshaler = (*DateTime)(nil)
+
 // MarshalJSON marshal to time type
 func (d DateTime) MarshalJSON() ([]byte, error) {
 	return json.Marshal(d.Time())
 }
 
+// UnmarshalJSON creates a primitive.DateTime from a JSON string.
+func (d *DateTime) UnmarshalJSON(data []byte) error {
+	// Ignore "null" to keep parity with the time.Time type and the standard library. Decoding "null" into a non-pointer
```

Comment:
Relevant sections from the `json.Unmarshal` docs:

1. If the JSON value is `null` and we're unmarshalling into a pointer, the code will never enter this function.
```
To unmarshal JSON into a pointer, Unmarshal first handles the case of the JSON being the JSON literal null. In that case, 
Unmarshal sets the pointer to nil. Otherwise, Unmarshal unmarshals the JSON into the value pointed at by the pointer. If 
the pointer is nil, Unmarshal allocates a new value for it to point to.
```

2. The special handling for `null` in this function only happens when unmarshalling into a value, not a pointer, and is in line with the stdlib behavior:
```
The JSON null value unmarshals into an interface, map, pointer, or slice by setting that Go value to nil. Because null is 
often used in JSON to mean “not present,” unmarshaling a JSON null into any other Go type has no effect on the value 
and produces no error.
```